src/lib/core/WeaveMessageLayer.cpp

```diff
@@ -2530,5 +2531,35 @@ void WeaveMessageLayer::GetPeerDescription(char * buf, size_t bufSize, const Wea
         msgInfo->InCon);
 }
 
+/**
+ * @brief
+ *   Generate random Weave device Id.
+ *
+ * @details
+ *   This function generates 64-bit locally unique Weave device Id. This function uses cryptographically strong
+ *   random data source to guarantee uniqueness of generated value. Note that bit 57 of the generated Weave
+ *   device Id is set to 1 to indicate that generated Weave device Id is locally (not globally) unique.
+ *
+ * @param deviceId                      A reference to the 64-bit Weave device Id.
+ *
+ * @retval  #WEAVE_NO_ERROR             If Weave device Id was successfully generated.
+ */
+NL_DLL_EXPORT WEAVE_ERROR GenerateWeaveDeviceId(uint64_t & deviceId)
+{
+    WEAVE_ERROR err;
+    uint64_t id = 0;
+
+    while (id <= kMaxAlwaysLocalWeaveNodeId)
+    {
+        err = nl::Weave::Platform::Security::GetSecureRandomData(reinterpret_cast<uint8_t*>(&id), sizeof(id));
+        SuccessOrExit(err);
+    }
+
+    deviceId = id | kEUI64_UL_Local;
```

Comment:
Where is kEUI64_UL_Local defined? Is there a file missing from the PR?

Reply:
These EUI64 defines were part of a different PR and they are in the WeaveMessageLayer.h file.